### PR TITLE
Reworks

### PR DESCRIPTION
**Pull request by Dyrati**
Added a 4-bit arrow_inputs property to the data object.  Right = 1, Left = 2, Up = 4, Down = 8.  It's currently not used, however you can obtain the direction from this by using data.arrow_inputs to index rotation_key, which I added to utils.js.  I also modified the direction changing section of movement.js.

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -22,6 +22,50 @@ export const reverse_directions = {
...
+    3,    //195-225 degrees
+    3,    //225-255 degrees
+    2,    //255-285 degrees
+    1,    //285-315 degrees
+    1,    //315-345 degrees
+]
+
+export function update_arrow_inputs(data) {
+    data.arrow_inputs = 
+        1 & data.cursors.right.isDown
```
Comment (jjppof) [inline utils.js:63]: Just changed `data.cursors.RIGHT` to `data.cursors.right`. It's lowercase.

**Comment by jjppof** [comment]
wrong speed
The speed when going in diagonal is reducing.

**Comment by jjppof** [comment]
weird move
There's a weird move happening depending on how you collide.

**Comment by jjppof** [comment]
jitter
Another weird thing + jitter.

**Comment by jjppof** [comment]
not_changing
Not changing the direction when colliding in diagonal.

File: utils.js [hunk 2]
```diff
@@ -22,6 +22,50 @@ export const reverse_directions = {
...
+    1,    //285-315 degrees
+    1,    //315-345 degrees
+];
+
+export function update_arrow_inputs(data) {
+    data.arrow_inputs =
+        1 & data.cursors.right.isDown
+        | 2 & data.cursors.left.isDown
+        | 4 & data.cursors.up.isDown
+        | 8 & data.cursors.down.isDown;
```
Comment (jjppof) [inline utils.js:66]: It seems that there's something wrong with this. This var only changes when pressing right.

Comment (jjppof) [reply utils.js:66]: Maybe `*` instead of `&` ?

Comment (jjppof) [reply utils.js]: Yeah, changing this worked.

Comment (Dyrati) [reply utils.js]: Ohhh, I see.  2 & true evaluates to zero, because true gets converted to 1 prior to the comparison.

**Pull request merged**